File: Day41/LongestCommonPrefix.java
import java.util.*;

public class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) return "";
        String prefix = strs[0];
        for (int i = 1; i < strs.length; i++) {
            while (!strs[i].startsWith(prefix)) {
                prefix = prefix.substring(0, prefix.length() - 1);
                if (prefix.isEmpty()) return "";
            }
        }
        return prefix;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine().replaceAll("[\\[\\]\"]", "");
        String[] strs = input.split(",");
        for (int i = 0; i < strs.length; i++) strs[i] = strs[i].trim();
        LongestCommonPrefix sol = new LongestCommonPrefix();
        System.out.println(sol.longestCommonPrefix(strs));
    }
}

File: Day41/LongestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        if not strs:
            return ""
        prefix = strs[0]
        for s in strs[1:]:
            while not s.startswith(prefix):
                prefix = prefix[:-1]
                if not prefix:
                    return ""
        return prefix

    @staticmethod
    def main():
        import ast
        line = input().strip()
        strs = ast.literal_eval(line)
        sol = Solution()
        print(sol.longestCommonPrefix(strs))

if __name__ == "__main__":
    Solution.main()
